tutorials/radiant-mlhub-on-demand-training-data.ipynb

```diff
@@ -0,0 +1,1281 @@
+   "id": "1eddda32-d50d-413d-add8-dedaa2f9a067",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def temporal_buffer(item_datetime: str, date_delta: int) -> str:\n",
+    "    \"\"\"Takes a datetime string and returns a buffer around that date\n",
+    "\n",
+    "    Args:\n",
+    "        item_datetime: string of the datetime property from an Item\n",
+    "        date_delta: integer for days to add before and after a date\n",
+    "\n",
+    "    Returns:\n",
+    "        a string range representing the full date buffer\n",
+    "    \"\"\"\n",
+    "    delta = td(days=date_delta)\n",
+    "    item_dt = dt.strptime(item_datetime, \"%Y-%m-%dT%H:%M:%SZ\")\n",
+    "\n",
+    "    dt_start = item_dt - delta\n",
+    "    dt_start_str = dt_start.strftime(\"%Y-%m-%d\")\n",
+    "\n",
+    "    dt_end = item_dt + delta\n",
+    "    dt_end_str = dt_end.strftime(\"%Y-%m-%d\")\n",
+    "\n",
+    "    return f\"{dt_start_str}/{dt_end_str}\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f77e4ec0-a1b8-490c-b732-4c10d89b06ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def min_cloud_cover_scene(label_geom: Polygon, search_items: ItemCollection) -> Item:\n",
+    "    \"\"\"Finds the Item with minimal cloud cover from an ItemCollection\n",
+    "\n",
+    "    Args:\n",
+    "        label_geom: Polygon geometry to ensure label completely within scene\n",
+    "        search_items: ItemCollection of the Items found from pystac_client search\n",
+    "\n",
+    "    Returns:\n",
+    "        Item where label completely contained within, and minimal cloud cover\n",
+    "    \"\"\"\n",
+    "    min_cc = np.inf\n",
+    "    min_cc_item = None\n",
+    "    for item in search_items:\n",
+    "        item_geom = shape(item.geometry)\n",
+    "        item_cc = item.properties[\"eo:cloud_cover\"]\n",
+    "        if item_cc < min_cc and label_geom.within(item_geom):\n",
+    "            min_cc = item_cc\n",
+    "            min_cc_item = item\n",
+    "    return min_cc_item"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c3b23fa5-d588-42d4-b913-fab7819e7ea3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_landsat_8_match(label_item: Item) -> Tuple[Item, Item]:\n",
+    "    \"\"\"Finds the best Landsat 8 match using source Item datetime and bounding box.\n",
+    "\n",
+    "    Args:\n",
+    "        label_item: the STAC label Item object\n",
+    "\n",
+    "    Returns:\n",
+    "        Tuple of the BigEarthNet source Item and the Landsat 8 match Item\n",
+    "    \"\"\"\n",
+    "    # get the matching source Item properties\n",
+    "    source_items = search_stac_api(\n",
+    "        catalog_client=mlhub_catalog,\n",
+    "        collections=[BIGEARTHNET_SOURCE_COLLECTION],\n",
+    "        ids=get_source_item_ids(label_item),\n",
+    "    )\n",
+    "\n",
+    "    if source_items:\n",
```

Comment:
Is there any reason for this if / elif? Are there cases where you don't have a match?